File: android/app/src/main/java/com/vernu/sms/database/local/AppDatabase.java
//package com.vernu.sms.database.local;
//
//import android.content.Context;
//import androidx.room.Database;
//import androidx.room.Room;
//import androidx.room.RoomDatabase;
//
//@Database(entities = {SMS.class}, version = 2)
//public abstract class AppDatabase extends RoomDatabase {
//    private static volatile AppDatabase INSTANCE;
//
//    public static AppDatabase getInstance(Context context) {
//        if (INSTANCE == null) {
//            synchronized (AppDatabase.class) {
//                if (INSTANCE == null) {
//                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class, "db1")
//                            .build();
//                }
//            }
//        }
//        return INSTANCE;
//    }
//
//    public abstract SMSDao localReceivedSMSDao();
//}

File: android/app/src/main/java/com/vernu/sms/database/local/AppDatabase.kt
package com.vernu.sms.database.local

/*
import android.content.Context
import androidx.room.*

@Database(entities = [Sms::class], version = 2)
@TypeConverters(DateConverter::class)
abstract class AppDatabase : RoomDatabase() {

    abstract fun localReceivedSMSDao(): SmsDao

    companion object {
        @Volatile
        private var INSTANCE: AppDatabase? = null

        fun getInstance(context: Context): AppDatabase {
            return INSTANCE ?: synchronized(this) {
                INSTANCE ?: Room.databaseBuilder(
                    context.applicationContext,
                    AppDatabase::class.java,
                    "db1"
                ).build().also { INSTANCE = it }
            }
        }
    }
}
*/
